traffic_ops/traffic_ops_golang/federations/ds.go

```diff
@@ -104,3 +105,37 @@ func getFedNameByID(tx *sql.Tx, id int) (string, bool, error) {
 	}
 	return name, true, nil
 }
+
+// TOFedDSes data structure to use on read of federation deliveryservices
+type TOFedDSes struct {
+	api.APIInfoImpl `json:"-"`
+	tc.FederationDeliveryServiceNullable
+}
+
+func (v *TOFedDSes) NewReadObj() interface{} { return &tc.FederationDeliveryServiceNullable{} }
+func (v *TOFedDSes) SelectQuery() string     { return selectQuery() }
+func (v *TOFedDSes) ParamColumns() map[string]dbhelpers.WhereColumnInfo {
+	return map[string]dbhelpers.WhereColumnInfo{
+		"id":   dbhelpers.WhereColumnInfo{"fds.federation", api.IsInt},
+		"dsID": dbhelpers.WhereColumnInfo{"fds.deliveryservice", api.IsInt},
+	}
+}
+func (v *TOFedDSes) GetType() string {
+	return "federation_deliveryservice"
+}
+
+func (v *TOFedDSes) Read() ([]interface{}, error, error, int) { return api.GenericRead(v) }
```

Comment:
I don't believe the Perl ever considered tenancy, so  won't demand that you do in the rewrite, but I wanted to point out that you could, if you so chose.

Reply:
Do you think it is better as part of the port from perl to golang? Or as part of a different effort to add tenancy?

Reply:
Either way, as you please.

Reply:
Ill defer to another PR